hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -204,6 +204,18 @@ public BitmapIndexOptions getBitmapIndexOptions() {
         return bitmapIndexOptions;
     }
 
+    /**
+     * Sets bitmap index options of this index config to the given ones.
+     *
+     * @param bitmapIndexOptions the bitmap index options to set.
+     * @return this index config instance.
+     */
+    public IndexConfig setBitmapIndexOptions(BitmapIndexOptions bitmapIndexOptions) {
+        this.bitmapIndexOptions.setUniqueKey(bitmapIndexOptions.getUniqueKey());
```

Comment:
These two lines look dangerous to me because we mutate an object which was set earlier. Consider the following code:
```
IndexConfig config = new IndexConfig();
BitmapIndexOptions options = config1.getBitmapIndexOptions();
// ...
config.setBitmapIndexOptions(<new options>);
// "options" object changed unexpectedly
```

I would rather overwrite an object in the setter and perform a lazy initialization in getter as we do in other config classes (e.g. `IndexConfig.getAttributes()` copies this approach).

Reply:
fixed in https://github.com/hazelcast/hazelcast/pull/16555/commits/656e85640ced1de138c37f08fde92b6fd624d8cf